ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java

```diff
@@ -788,6 +833,286 @@ private void migrateToPropertyStore() throws InterruptedException {
+              .stream()
+              .filter(s -> s.matches("\\d+"))
+              .collect(Collectors.toSet());
+          Set<String> resources;
+          if (commaSeparatedResources.equalsIgnoreCase(ALL)) {
+            resources = helixResources;
+          } else {
+            resources = Arrays.stream(commaSeparatedResources.replaceAll("\\p{Space}", "").split(","))
+                .collect(Collectors.toSet());
+            resources.removeIf(resource -> {
+              if (!helixResources.contains(resource)) {
+                logger.info("Resource {} is not present in data center {}", resource, dcName);
+                return true;
+              }
+              return false;
+            });
+          }
+
+          Map<String, IdealState> resourceToIdealState = new HashMap<>();
+          for (String resource : resources) {
+            resourceToIdealState.put(resource, helixAdmin.getResourceIdealState(clusterName, resource));
+          }
+
+          ConfigAccessor configAccessor =
+              new ConfigAccessor(dataCenterToZkAddress.get(dcName).getZkConnectStrs().get(0));
+
+          // 1. Update cluster config
+          setupCluster(configAccessor);
+
+          // 2. Update instance config for all instances present in each resource
+          Set<String> allInstances = new HashSet<>();
+          for (String resource : resources) {
+            // a. Get list of instances sharing partitions in this resource
+            IdealState idealState = resourceToIdealState.get(resource);
+            Set<String> partitions = idealState.getPartitionSet();
+            Set<String> instances = new HashSet<>();
+            for (String partition : partitions) {
+              instances.addAll(idealState.getInstanceSet(partition));
+            }
+            // b. Update instance config for each instance
+            for (String instance : instances) {
+              setupInstanceConfig(dcName, instance, resource, configAccessor, propertyStoreAdapter);
+            }
+            allInstances.addAll(instances);
+          }
+
+          // 3. No resource config to update for now
+
+          // 4. Validate configs
+          Map<String, Boolean> instanceValidationResultMap =
+              helixAdmin.validateInstancesForWagedRebalance(clusterName, new ArrayList<>(allInstances));
+          if (instanceValidationResultMap.containsValue(Boolean.FALSE)) {
+            throw new IllegalStateException("Instances are not configured correctly for waged rebalancer");
+          }
+          Map<String, Boolean> resourceValidationResultMap =
+              helixAdmin.validateResourcesForWagedRebalance(clusterName, new ArrayList<>(resources));
+          if (resourceValidationResultMap.containsValue(Boolean.FALSE)) {
+            throw new IllegalStateException("Resources are not configured correctly for waged rebalancer");
+          }
+
+          if (!dryRun) {
```

Comment:
let's print out some log message saying this is not a dryrun.

Reply:
Sure. Added log messages